### ISSUE TITLE
Parse Error on Hash with quoted key and colon syntax

### ISSUE THREAD

**Issue by daveknapik** [presidentbeef/brakeman#1331]
### Background

Brakeman version: 4.5.0
Rails version: 5.2.2
Ruby version: 2.5.1

Link to Rails application code: 

private repo at work :(

#### Parse Error

```
$ ruby_parse lib/classes/cross_sell_email/project_data/kyoto_mie.rb

# file = lib/classes/cross_sell_email/project_data/kyoto_mie.rb loc = 49
/Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/racc/parser.rb:528:in `on_error' #<Racc::ParseError: lib/classes/cross_sell_email/project_data/kyoto_mie.rb:4 :: parse error on value "[" (tLBRACK2)> for lib/classes/cross_sell_email/project_data/kyoto_mie.rb
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/racc/parser.rb:528:in `on_error'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/lib/ruby_parser_extras.rb:1162:in `on_error'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/racc/parser.rb:259:in `_racc_do_parse_c'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/racc/parser.rb:259:in `do_parse'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/lib/ruby_parser_extras.rb:1069:in `block in process'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/timeout.rb:93:in `block in timeout'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/timeout.rb:33:in `block in catch'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/timeout.rb:33:in `catch'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/timeout.rb:33:in `catch'
  /Users/daveknapik/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/timeout.rb:108:in `timeout'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/lib/ruby_parser_extras.rb:1057:in `process'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/lib/ruby_parser.rb:35:in `block in process'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/lib/ruby_parser.rb:32:in `each'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/lib/ruby_parser.rb:32:in `process'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/bin/ruby_parse:48:in `block in <top (required)>'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/bin/ruby_parse:29:in `each'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/gems/ruby_parser-3.13.0/bin/ruby_parse:29:in `<top (required)>'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/bin/ruby_parse:23:in `load'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/bin/ruby_parse:23:in `<main>'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/bin/ruby_executable_hooks:24:in `eval'
  /Users/daveknapik/.rvm/gems/ruby-2.5.1@voyagin/bin/ruby_executable_hooks:24:in `<main>'
done


 0.01s:  6429.60 l/s:  286.52 Kb/s:    2 Kb:   49 loc:lib/classes/cross_sell_email/project_data/kyoto_mie.rb

 0.01s:  6429.60 l/s:  286.52 Kb/s:    2 Kb:   49 loc:TOTAL
```

Minimal example that does not parse:

```ruby
ACT_IDS = {
  "#{SECTION_SPONSORED_CONTENT_WITH_ACTIVITY_CARDS}": [ 10402, 10403, 10489 ],
}
```

If I change that to hashrocket syntax, it is happy, but AFAIK, the way I have it is legitimate Ruby code, as it works on my console. Thanks.

**Comment by presidentbeef** [comment]
Thank you for reporting!

I have reduced the example down to:

```ruby
{ "#{}": [] }
```

The interpolation and the array value seem to be required for the parse failure.

@zenspider ^

**Comment by daveknapik** [comment]
No problem, thank you for verifying the behavior and the prompt reply! 😄

**Comment by bermannoah** [comment]
Having the same issue, glad to see it was already reported! :) If it helps to know it happens on Rails 4.2.11.1 and Ruby 2.4.5 as well.

**Comment by zenspider** [comment]
Fixed

**Comment by presidentbeef** [comment]
This was fixed in ruby_parser 3.13.1 which is used by Brakeman 4.5.1.
